supervisor/api/network.py

```diff
@@ -0,0 +1,81 @@
+"""REST API for network."""
+from typing import Any, Dict
+
+from aiohttp import web
+import voluptuous as vol
+
+from ..const import (
+    ATTR_GATEWAY,
+    ATTR_ID,
+    ATTR_INTERFACES,
+    ATTR_IP_ADDRESS,
+    ATTR_PRIMARY,
+    ATTR_TYPE,
+)
+from ..coresys import CoreSysAttributes
+from .utils import api_process, api_validate
+
+SCHEMA_UPDATE = vol.Schema(
+    {
+        vol.Optional("address"): vol.Coerce(str),
+        vol.Optional("mode"): vol.Coerce(str),
+        vol.Optional("gateway"): vol.Coerce(str),
+        vol.Optional("dns"): vol.Coerce(str),
+    }
+)
+
+
+class APINetwork(CoreSysAttributes):
+    """Handle REST API for network."""
+
+    @api_process
+    async def info(self, request: web.Request) -> Dict[str, Any]:
+        """Return network information."""
+        interfaces = {}
+        for interface in self.sys_dbus.network.interfaces:
+            interfaces[self.sys_dbus.network.interfaces[interface].name] = {
+                ATTR_IP_ADDRESS: self.sys_dbus.network.interfaces[interface].ip_address,
+                ATTR_GATEWAY: self.sys_dbus.network.interfaces[interface].gateway,
+                ATTR_ID: self.sys_dbus.network.interfaces[interface].id,
+                ATTR_TYPE: self.sys_dbus.network.interfaces[interface].type,
+                ATTR_PRIMARY: self.sys_dbus.network.interfaces[interface].primary,
```

Comment:
you can use `as_dict` from attrs if you like

Reply:
Yeah, but I also have to do an "on the fly" conversion, and with this approach, there is more control to what's show.
Since it's used in multiple API responses I have extracted the dict generation to a function that is called with both API's